### ISSUE TITLE
Error parsing script 'check'

### ISSUE THREAD

**Issue by gaoxiaoliangz** [denoland/deno#20332]
I have this in my deno.json
```
{
  "tasks": {
    "check": "deno check services/*/{prod,dev}.ts"
  }
```
When I run `deno task check`, I got:
```
Task check deno check services/*/{prod,dev}.ts
error: Error parsing script 'check'.

Caused by:
    Unexpected character.
      {prod,dev}.ts
```

My deno version:
```
deno 1.36.3 (release, aarch64-apple-darwin)
v8 11.6.189.12
typescript 5.1.6
```

**Comment by bartlomieju** [comment]
@dsherret is this supposed to work?

**Comment by dsherret** [comment]
No, [brace expansion](https://www.gnu.org/software/bash/manual/html_node/Brace-Expansion.html) has not been implemented in deno task.

**Comment by armichaud** [comment]
This is something I'd be interested in taking a crack at. Looking at the GNU docs @dsherret shared, there are a few discrete features of brace expansion we might consider implementing.

1. Basic: Comma separate string lists, e.g. `"deno check services/*/{prod,dev}.ts"`
2. Nested: `"deno check services/{service1/{prod,staging,dev},service2/{prod,dev}}.ts"`
3. Numeric Range Expansion: `"deno check services/service{0..10[..2]}/{prod,dev}.ts"`
4. Lexicographical Range Expansion: `"deno check services/service{a..z}/{prod,dev}.ts"`

Should I aim to tackle all four at once? Or should we start perhaps with just the first 1 or 2 and go from there?

**Comment by armichaud** [comment]
Having looked a bit further into this, I think this would necessitate changes to the `deno_task_shell`. @dsherret should we transfer this issue there?
